### PR TITLE
Add ERC: On-chain NFT royalty enforcement

### PR DESCRIPTION
**Pull request by ARR4N**
This was originally [PR #7781 of the EIPs repo](https://github.com/ethereum/EIPs/pull/7781), which was closed because of the migration to this repo. The ERC number 7526 was assigned by @abcoathup in that PR.

### CODE REVIEW

**Comment by xinbenlv** [comment]
Hi, the interface in specification section has not been finished, can you finish them before you turn this PR for merge again?

**Comment by aschlosberg** [comment]
> Hi, the interface in specification section has not been finished, can you finish them before you turn this PR for merge again?

Shall do. Once done I'll tag you and move the PR out of draft.

**Comment by aschlosberg** [comment]
I've added the interface, @xinbenlv. Please can you take another look?

**Comment by aschlosberg** [comment]
@axic @g11tech @SamWilsn @xinbenlv I think this has fallen through the cracks. Can we please get a review?

I appreciate that some of the delay was due to the repo migration, but it's been 6 months since we first submitted.

File: ERCS/erc-7526.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+eip: 7526
+title: On-chain NFT royalty enforcement
+description: A game-theoretic mechanism to induce truthful revelation of NFT sales and their value to allow for decentralised royalty enforcement.
```
Comment (xinbenlv) [inline ERCS/erc-7526.md:4]:
```suggestion
description: An interface for game-theoretic mechanism to induce truthful revelation of NFT sales and their value to allow for decentralised royalty enforcement.
```


ERCs are standard track: most ERCs are used to propose a way for interoperability. Using it to propose a "mechanism" is a bit confusing. Maybe add "interface" in your summary.

Also, I don't see game theory being used other than the motivation. Can you show it in the reference implementation of what you refer to?

File: ERCS/erc-7526.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+This problem is aggravated in over-the-counter (OTC) deals because the flow of funds can be handled entirely via hidden channels.
+
+However, on-chain royalty enforcement can still be feasible in practice by creating a sufficiently strong incentive or penalty to force participating parties to provide honest information.
+We describe one such approach, a game-theoretic mechanism that targets the primary actors in a transfer (i.e. the previous and new token owner) as the only players, and induce the new owner to truthfully supply the information required for enforcement.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+For each `tokenId` we introduce a set of addresses whose members MAY transfer the token to themselves without payment by calling `takeBack(uint256 tokenId)`, which we will refer to as a *take-back*.
```
Comment (xinbenlv) [inline ERCS/erc-7526.md:35]:
```suggestion
For each `tokenId` we introduce a set of addresses whose members MAY transfer the token to themselves without payment by calling `takeBack(uint256 tokenId)`, which we will refer to as a *take-back*.
```
"we introduce a set of addresses": can you ellaborate?

File: ERCS/erc-7526.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+Marketplaces facilitating sales SHOULD forward royalties in the same transaction as the sale to guarantee payment within the grace period, thus immediately unfreezing the token.
+Marketplaces MAY allow buyers to configure either the auto-listing price or the royalties paid (as the outcome is equivalent), or they MAY provide sensible default values or employ more sophisticated heuristics at their discretion.
+If allowing buyers to configure values, marketplaces MAY limit this to the auto-listing price for ease of understanding as royalties are then conceptually equivalent to ERC-2981's `royaltyInfo(autoListingPrice)`, which is a concept familiar to end users.
+
+```solidity
+/**
+ * @dev This constitutes a rough draft of the minimum required functions, and will almost certainly change as we
+ * complete the reference implementation.
+ */
```
Comment (SamWilsn) [inline ERCS/erc-7526.md:67]: Proposals should be written as if they are already complete and ready to use.

If you want to leave yourself notes, you can use HTML-style comments (`<!-- TODO -->`) though they'll have to be outside of the code fence.

File: ERCS/erc-7526.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+* Accepting the risk of take-back in the event of no royalty being paid is only a rational decision if the same entity controls both addresses; i.e. a change in beneficial owner of the token results in a royalty payment but transfer between one's own addresses doesn't require one.
+* The auto-listing window ensures *truthful* revelation of the value of the token.
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Reference Implementation
+
+TBD
```
Comment (SamWilsn) [inline ERCS/erc-7526.md:120]:
```suggestion
TBD <!-- TODO -->
```

File: ERCS/erc-7526.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+## Security Considerations
+
+Needs discussion.
```
Comment (SamWilsn) [inline ERCS/erc-7526.md:124]:
```suggestion
Needs discussion. <!-- TODO -->
```

File: ERCS/erc-7526.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          [2024, 2, 19]
+        ]
+      },
+      "URL": "https://arxiv.org/abs/2403.07896",
+      "custom": {
+        "additional-urls": [
+        ]
+      }
+    }
+    ```
```
Comment (SamWilsn) [inline ERCS/erc-7526.md:107]: If you hold the copyrights to the paper, you can instead include it in the `assets/` directory.

File: ERCS/erc-7526.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+### Summary rationale for the draft:
```
Comment (SamWilsn) [inline ERCS/erc-7526.md:109]: Again, write as if this was the final version.

**Pull request closed without merging**